src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,59 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private static final Map<Class<?>, Class<?>> interfaceImplementations = new HashMap<>();
```

Comment:
According to the checklist, the Interface Implementations map should be a class field and filled using Map.of(). Consider using Map.of() instead of new HashMap<>(), as it is more concise and immutable.